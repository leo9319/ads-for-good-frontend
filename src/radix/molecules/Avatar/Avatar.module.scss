@import '../../../styles/colours';
@import './AvatarCustomStyle';

.avatarContainer {
  cursor: pointer;
  // size 1-5 avatar size token
  --space-5: var(--spacing-lg) !important;
  --space-6: 36px !important;
  --space-7: 48px !important;
  --space-8: var(--spacing-4xl) !important;
  --space-9: var(--spacing-5xl) !important;

  > span {
    background-color: var(--avatarBackgroundColor) !important;
    color: var(--avatarTextColor) !important;
    overflow: hidden;
  }
}
