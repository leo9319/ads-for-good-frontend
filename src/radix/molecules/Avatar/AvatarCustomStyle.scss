@import '../../../styles/breakpoints';
@import '../../../styles/spacing';

.avatarContainer {
  &.rt-r-size-1 {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-1);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-md);
        height: var(--spacing-md);
      }
    }
  }
  &.rt-r-size-2 {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-2);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg);
        height: var(--spacing-lg);
      }
    }
  }
  &.rt-r-size-3 {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-4);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg);
        height: var(--spacing-lg);
      }
    }
  }
  &.rt-r-size-4 {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-6);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg);
        height: var(--spacing-lg);
      }
    }
  }
  &.rt-r-size-5 {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-7);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg);
        height: var(--spacing-lg);
      }
    }
  }
  &.rt-r-size-6 {
    --avatar-size: 96px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg);
        height: var(--spacing-lg);
      }
    }
  }
  &.rt-r-size-7 {
    --avatar-size: 112px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl);
        height: var(--spacing-2xl);
      }
    }
  }
  &.rt-r-size-8 {
    --avatar-size: 144px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8);
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl);
        height: var(--spacing-2xl);
      }
    }
  }
  &.rt-r-size-9 {
    --avatar-size: 160px;
    .rt-AvatarFallback {
      font-size: 40px;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-3xl);
        height: var(--spacing-3xl);
      }
    }
  }
}

// xs 1-9

@media (min-width: $xs-min) {
  .avatarContainer:where(.xs\:rt-r-size-1) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-1) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-md) !important;
        height: var(--spacing-md) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-2) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-2) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-3) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-4) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-4) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-6) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-5) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-7) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-6) {
    --avatar-size: 96px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-7) {
    --avatar-size: 112px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-8) {
    --avatar-size: 144px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.xs\:rt-r-size-9) {
    --avatar-size: 160px;
    .rt-AvatarFallback {
      font-size: 40px !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-3xl) !important;
        height: var(--spacing-3xl) !important;
      }
    }
  }
}

// sm 1-9

@media (min-width: $sm-min) {
  .avatarContainer:where(.sm\:rt-r-size-1) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-1) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-md) !important;
        height: var(--spacing-md) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-2) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-2) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-3) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-4) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-4) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-6) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-5) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-7) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-6) {
    --avatar-size: 96px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-7) {
    --avatar-size: 112px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-8) {
    --avatar-size: 144px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.sm\:rt-r-size-9) {
    --avatar-size: 160px;
    .rt-AvatarFallback {
      font-size: 40px !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-3xl) !important;
        height: var(--spacing-3xl) !important;
      }
    }
  }
}

// md 1-9

@media (min-width: $md-min) {
  .avatarContainer:where(.md\:rt-r-size-1) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-1) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-md) !important;
        height: var(--spacing-md) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-2) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-2) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-3) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-4) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-4) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-6) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-5) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-7) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-6) {
    --avatar-size: 96px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-7) {
    --avatar-size: 112px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-8) {
    --avatar-size: 144px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.md\:rt-r-size-9) {
    --avatar-size: 160px;
    .rt-AvatarFallback {
      font-size: 40px !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-3xl) !important;
        height: var(--spacing-3xl) !important;
      }
    }
  }
}

// lg 1-9

@media (min-width: $lg-min) {
  .avatarContainer:where(.lg\:rt-r-size-1) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-1) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-md) !important;
        height: var(--spacing-md) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-2) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-2) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-3) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-4) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-4) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-6) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-5) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-7) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-6) {
    --avatar-size: 96px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-7) {
    --avatar-size: 112px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-8) {
    --avatar-size: 144px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.lg\:rt-r-size-9) {
    --avatar-size: 160px;
    .rt-AvatarFallback {
      font-size: 40px !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-3xl) !important;
        height: var(--spacing-3xl) !important;
      }
    }
  }
}

// xl 1-9

@media (min-width: $xl-min) {
  .avatarContainer:where(.xl\:rt-r-size-1) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-1) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-md) !important;
        height: var(--spacing-md) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-2) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-2) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-3) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-4) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-4) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-6) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-5) {
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-7) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-6) {
    --avatar-size: 96px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-lg) !important;
        height: var(--spacing-lg) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-7) {
    --avatar-size: 112px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-8) {
    --avatar-size: 144px;
    .rt-AvatarFallback {
      font-size: var(--typography-font-size-8) !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-2xl) !important;
        height: var(--spacing-2xl) !important;
      }
    }
  }

  .avatarContainer:where(.xl\:rt-r-size-9) {
    --avatar-size: 160px;
    .rt-AvatarFallback {
      font-size: 40px !important;
    }
    .avatarIcon {
      .icon {
        width: var(--spacing-3xl) !important;
        height: var(--spacing-3xl) !important;
      }
    }
  }
}
