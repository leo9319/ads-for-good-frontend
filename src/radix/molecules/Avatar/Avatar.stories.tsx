import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import Avatar from './Avatar';
import { Theme } from '@radix-ui/themes';

const meta = {
  title: 'Library/Molecules/Avatar',
  component: Avatar,
  parameters: {
    layout: 'center',
  },
  tags: ['autodocs'],
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  argTypes: {
    size: {
      control: 'select',
      options: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
    },
    src: {
      control: 'text',
    },
    textFallback: {
      control: 'text',
    },
    backgroundColor: {
      control: 'color',
    },
    textColor: {
      control: 'color',
    },
    radius: {
      control: 'select',
      options: ['none', 'small', 'medium', 'large', 'full'],
    },
    fallback: {
      control: 'text',
    },
  },
} satisfies Meta<typeof Avatar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const AvatarWithProfile: Story = {
  args: {
    size: '2',
    src: 'https://www.gstatic.com/webp/gallery/1.jpg',
    backgroundColor: '#E86100',
    textColor: '#fff',
    radius: 'full',
    textFallback: 'WV',
    fallback: 'icon-user',
  },
};

export const AvatarWithInitialsAccent: Story = {
  args: {
    size: '2',
    backgroundColor: '#E86100',
    textColor: '#fff',
    radius: 'full',
    textFallback: 'WV',
    fallback: '',
  },
};

export const AvatarWithIconAccent: Story = {
  args: {
    size: '2',
    backgroundColor: '#E86100',
    textColor: '#fff',
    radius: 'full',
    textFallback: '',
    fallback: 'icon-user',
  },
};

export const AvatarWithIconDefault: Story = {
  args: {
    size: '2',
    backgroundColor: '#F4F5F7',
    textColor: '#65676B',
    radius: 'full',
    textFallback: '',
    fallback: 'icon-user',
  },
};

export const AvatarWithInitialsDefault: Story = {
  args: {
    size: '2',
    backgroundColor: '#F4F5F7',
    textColor: '#65676B',
    radius: 'full',
    textFallback: 'WV',
    fallback: '',
  },
};
