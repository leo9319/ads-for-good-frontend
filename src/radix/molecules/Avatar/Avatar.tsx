import React from 'react';
import { Avatar as RadixAvatar, AvatarProps as Props } from '@radix-ui/themes';
import { classNames, getModuleClasses } from '@utils/common/classNames';
import { getStylesAsCssProperties } from '@utils/common/styles';
import { AvatarSizes } from '@internal/types/common/style';
import Icons from '@components/atoms/Icons';

import './AvatarCustomStyle.scss';
import styles from './Avatar.module.scss';

/**
 * @interface
 * The Avatar props type of {@link Avatar}
 *
 * @property {Props} props - The props of the Avatar component
 */
// export type AvatarProps = Props;

export type AvatarProps = Props & {
  /**
   * The size of the Avatar
   * @property
   */
  size?: AvatarSizes;
  /**
   * The initial fallback text of the Avatar
   * @property
   */
  textFallback?: string;
  /**
   * The Background color of the Avatar
   * @property
   */
  backgroundColor?: string;
  /**
   * The Text color of the Avatar
   * @property
   */
  textColor?: string;
  /**
   * The alt text  of the Avatar
   * @property
   */
  altText?: string;
  /**
     * Callback function triggered when the Avatar is clicked
     @property
    */
  onAvatarClick?: () => void;
};

/**
 *
 * Avatar component to display the Avatar bar.
 *
 * @group Atoms
 * @category Radix Component
 *
 * @param {AvatarProps} props - The Avatar props
 * @returns {ReactElement} - Avatar Component
 *
 */

export const Avatar = ({
  className,
  backgroundColor = '#F4F5F7',
  src,
  fallback = 'icon-user',
  textFallback,
  textColor = '#65676B',
  size = '2',
  radius = 'full',
  altText = 'Avatar',
  onAvatarClick,
  ...props
}: AvatarProps) => {
  const classes = getModuleClasses(className?.trim(), styles);
  const wrapperClass = classNames(
    styles.avatarContainer,
    'avatarContainer',
    classes
  );

  return (
    <RadixAvatar
      className={wrapperClass}
      style={getStylesAsCssProperties({
        '--avatarBackgroundColor': backgroundColor,
        '--avatarTextColor': textColor,
      })}
      size={size}
      src={src}
      radius={radius}
      alt={altText}
      fallback={
        textFallback ? (
          textFallback
        ) : (
          <Icons
            name={fallback as string}
            className={classNames(styles.avatarIcon, 'avatarIcon')}
          />
        )
      }
      onClick={onAvatarClick}
      {...props}
    ></RadixAvatar>
  );
};

export default Avatar;
