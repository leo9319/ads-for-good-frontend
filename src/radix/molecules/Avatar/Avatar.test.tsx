import React from 'react';
import Avatar from './Avatar';
import { render } from '@testing-library/react';

describe('Avatar', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Avatar fallback="" />);
    expect(baseElement).toMatchSnapshot();
  });
});
