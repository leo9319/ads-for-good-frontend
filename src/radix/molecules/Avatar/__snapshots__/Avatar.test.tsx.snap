// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render successfully 1`] = `
<body>
  <div>
    <span
      class="rt-reset rt-AvatarRoot rt-r-size-2 rt-variant-soft avatarContainer avatarContainer"
      data-accent-color=""
      data-radius="full"
      style="--avatarBackgroundColor: #F4F5F7; --avatarTextColor: #65676B;"
    />
  </div>
</body>
`;
